### ISSUE TITLE
CI: Migrate `cargo test` to `cargo-nextest`

### ISSUE THREAD

**Issue by cmichi** [paritytech/ink#1474]
## Motivation

[`cargo-nextest`](https://nexte.st) can be used instead of `cargo test`. It runs the tests as well, but is more performant. We should replace our usage of `cargo test` with it.

## ToDo

- [ ] Install `cargo-nextest` in the Docker image that use in our CI. The image is called `ink-ci-linux`, you can find the Dockerfile [here](https://github.com/paritytech/scripts/blob/master/dockerfiles/ink-ci-linux/Dockerfile). Just add a `cargo install` there.
- [ ] Once merged you need to rebuild the Docker image. This is done every night. You can also trigger it by hand [here](https://gitlab.parity.io/parity/mirrors/scripts/-/pipeline_schedules), by click the play button: 
<img width="1054" alt="Bildschirmfoto 2022-11-05 um 21 49 57" src="https://user-images.githubusercontent.com/241530/200140789-2a25abc0-37c8-4d79-8f36-3787315ffcb0.png">. If you don't see a play button, ask in the "ink! <> CI/CD" channel. If the last build failed (as in this picture), ask the current 🦖 for help.
- [ ] Once the Docker build succeeded you can create an ink! PR that replaces `cargo test` in our `.gitlab-ci.yml` with `cargo-nextest` invocations.
- [ ] Check if our GitHub actions under `.github/` also need to be migrated.

## Follow-up

In case this ticket worked out good and we don't see issues: please create a follow-up issue for `cargo-contract`.

**Comment by cmichi** [comment]
I'm closing this PR and the issue as the results are inconclusive, we might revisit after https://github.com/paritytech/ink/issues/1520.
